nameserver/dns.go

```diff
@@ -227,7 +227,9 @@ func (d *DNSServer) handleRecursive(client *dns.Client, defaultMaxResponseSize i
 		}
 
 		for _, server := range d.upstream.Servers {
-			response, _, err := client.Exchange(req, fmt.Sprintf("%s:%s", server, d.upstream.Port))
+			reqCopy := *req
```

Comment:
Ah good catch!  I was looking for it and I didn't find one.  Do you know where I could find package level docs for the dns stuff?